Runtime/Device/TouchPanel/AdxTouchPanel.cs

```diff
@@ -369,28 +299,87 @@ public override async Task Write<T>(params T[] interactions)
 
             foreach (var command in commandBytes)
             {
-                await _connection.Write(command);
+                await _connection.WriteAsync(command);
             }
         }
 
-        // source: https://stackoverflow.com/a/48599119
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static bool ByteArraysEqual(ReadOnlySpan<byte> a1, ReadOnlySpan<byte> a2)
+        ReadOnlySpan<int> GetPacketHeadIndexs(Span<int> buffer, ReadOnlySpan<byte> packet)

```

Comment:
What is the reasoning for making the logic more complex and less efficient? Was there issues with the previous implementation?